arangod/Aql/ClusterBlocks.cpp

```diff
@@ -1376,6 +1376,21 @@ SortingGatherBlock::SortingGatherBlock(
     _sortRegisters
   );
 }
+  
+SortingGatherBlock::~SortingGatherBlock() {
+  clearBuffers();
+}
+
+void SortingGatherBlock::clearBuffers() noexcept {
+  for (std::deque<AqlItemBlock*>& x : _gatherBlockBuffer) {
+    for (AqlItemBlock* y : x) {
+      delete y;
+    }
+    x.clear();
+  }
```

Comment:
Wouldn't it be simpler and safer to change the type of `_gatherBlockBuffer` from `std::vector<std::deque<AqlItemBlock*>>` to `std::vector<std::deque<std::unique_ptr<AqlItemBlock>>>` and let the default destructors handle it?

Reply:
I agree.
Should be save to use, it is handed over by reference to the Sorting implementations.